### DISCUSSION TITLE
How to get "raw" links from GitHub (getting the raw image link)

### DISCUSSION THREAD

**Discussion by mrdbourke** [mrdbourke/pytorch-deep-learning#530] [General]
# Getting the raw file from the images directory

Since GitHub recently changed their design, the "raw" option isn't a button anymore.

So you can now just get the link via going to image and right-clicking and then selecting "Copy Image Address" option.

Steps:
1. Go to original image (e.g. https://github.com/mrdbourke/pytorch-deep-learning/blob/main/images/04-pizza-dad.jpeg) 
2. Right click
3. Select "Copy Image Address" (should end with 'raw?=true', e.g. https://github.com/mrdbourke/pytorch-deep-learning/blob/main/images/04-pizza-dad.jpeg?raw=true)

<img width="1499" alt="pizza-dad-copy-image-address" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/16750345/0ce3e9dc-9666-41f0-baf9-d85b165acf52">

See the new code snippet:

```python
# You can get this by right clicking on the image at: https://github.com/mrdbourke/pytorch-deep-learning/blob/main/images/04-pizza-dad.jpeg 
# And selecting "copy image address"
image_address = "https://github.com/mrdbourke/pytorch-deep-learning/blob/main/images/04-pizza-dad.jpeg?raw=true"

# Download custom image
import requests
from pathlib import Path

# Setup path to data folder
data_path = Path("data/")
data_path.mkdir(exist_ok=True)

# Setup custom image path
custom_image_path = data_path / "04-pizza-dad.jpeg"

# Download the image if it doesn't already exist
if not custom_image_path.is_file():
    with open(custom_image_path, "wb") as f:
        # When downloading from GitHub, need to use the "raw" file link
        request = requests.get(image_address) ### <- CHANGED
        print(f"Downloading {custom_image_path}...")
        f.write(request.content)
else:
    print(f"{custom_image_path} already exists, skipping download.")

# Display image
from IPython.display import Image, display

display(Image(custom_image_path))
```

**Comment by bluenoe** [comment]
thank a lot!
